File: run_mcp_client.mjs
import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio.js";
import fs from 'fs';

async function run() {
    const transport = new StdioClientTransport({
        command: "npx.cmd",
        args: ["-y", "mcp-remote", "https://stitch.googleapis.com/mcp", "--header", "X-Goog-Api-Key: AQ.Ab8RN6INM7BWKgxxR6h9UpigowVEHqsOkRCmen_aQyOLdDTdyQ"]
    });

    const client = new Client(
        {
            name: "test-client",
            version: "1.0.0"
        },
        {
            capabilities: {}
        }
    );

    try {
        await client.connect(transport);
        console.log("Connected");

        const projectName = "Credit Risk Platform Redesign";
        const res = await client.callTool({
            name: "create_project",
            arguments: {
                title: projectName
            }
        });

        console.log(JSON.stringify(res, null, 2));
        fs.writeFileSync('stitch_create_result.json', JSON.stringify(res, null, 2));

        // Next: Generate screen from text
        const projectId = res.content.find(c => c.text)?.text; // Or parse projectId differently
        fs.writeFileSync('stitch_project.json', JSON.stringify({ projectId }));

    } catch (e) {
        console.error(e);
    } finally {
        process.exit(0);
    }
}

run();

File: run_mcp_client.py
import asyncio
from mcp.client.stdio import stdio_client
from mcp.client.session import ClientSession
import sys
import json

async def run(method, params=None):
    from mcp import StdioServerParameters
    server_params = StdioServerParameters(
        command="npx.cmd",
        args=["-y", "mcp-remote", "https://stitch.googleapis.com/mcp", "--header", "X-Goog-Api-Key: AQ.Ab8RN6INM7BWKgxxR6h9UpigowVEHqsOkRCmen_aQyOLdDTdyQ"]
    )

    async with stdio_client(server_params) as (read, write):
        async with ClientSession(read, write) as session:
            await session.initialize()
            print("Session initialized.", file=sys.stderr)

            if method == "list_tools":
                result = await session.list_tools()
                print(json.dumps(result.model_dump()))
            elif method == "call_tool":
                tool_name = params.get("name")
                tool_args = params.get("arguments", {})
                result = await session.call_tool(tool_name, tool_args)
                print(json.dumps(result.model_dump()))
            else:
                print(f"Unknown method {method}")

if __name__ == "__main__":
    method = sys.argv[1]
    params_str = sys.argv[2] if len(sys.argv) > 2 else "{}"
    params = json.loads(params_str)
    asyncio.run(run(method, params))
